api/william_resolver.go

```diff
@@ -0,0 +1,75 @@
+// https://github.com/topfreegames/maestro
+//
+// Licensed under the MIT license:
+// http://www.opensource.org/licenses/mit-license
+// Copyright © 2017 Top Free Games <backend@tfgco.com>
+
+package api
+
+import (
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/topfreegames/maestro/models"
+	"net/http"
+)
+
+type PermissionResolver interface {
+	ResolvePermission(app *App, req *http.Request) (permission, resource string, err error)
+}
+
+type actionResolver struct {
+	action string
+}
+
+func (r actionResolver) ResolvePermission(app *App, req *http.Request) (permission, resource string, err error) {
```

Comment:
Shouldn't the method receiver be a pointer?

Reply:
Yeah, will change that. Just forgot to put it there 🤦